Updated the starting pollution multiplier

Jayefuu opened the pull request:
The new meta sucks. No one builds any walls or defences. Players can go 3 or 4 hours with no attacks so long as they craft nothing.

Now that the pollution from outposts is generated at position = {0,0}.... let's try a very small amount of pollution for every outpost captured to force at least a small amount of defences to be built.

This will be MUCH less pollution than if all of the items they get free were made in base, but it will still force at least a few defenses or clearing of nearby bases.

Starting with a value of 10 to see how it goes.

JamesFWilliamson committed:
Updated the starting pollution multiplier

The new meta sucks. No one builds any walls or defences. Players can go 3 or 4 hours with no attacks so long as they craft nothing.

Now that the pollution from outposts is generated at position = {0,0}.... let's try a very small amount of pollution for every outpost captured to force at least a small amount of defences to be built.

This will be MUCH less pollution than if all of the items they get free were made in base, but it will still force at least a few defenses or clearing of nearby bases.

Starting with a value of 10 to see how it goes.

JamesFWilliamson committed:
Fixed failed lua check

grilledham commented on map_gen/maps/crash_site/outpost_builder.lua:
```diff
@@ -1227,6 +1228,11 @@ local function set_pollution_multiplier(args, player)
     end
 
+    if multiplier < base_pollution_multiplier then
+        pollution_multiplier.value = base_pollution_multiplier
+        player.print("Cannot set the multiplier lower than the base multiplier of " .. base_pollution_multiplier)
```
Missing a `return`?
```suggestion
        pollution_multiplier.value = base_pollution_multiplier
        player.print("Cannot set the multiplier lower than the base multiplier of " .. base_pollution_multiplier)
        return
```

Jayefuu replied:
Yis. Fixed.

JamesFWilliamson committed:
Missing return

grilledham commented on map_gen/maps/crash_site/outpost_builder.lua:
```diff
@@ -1227,6 +1228,12 @@ local function set_pollution_multiplier(args, player)
     end
 
+    if multiplier < base_pollution_multiplier then
+        pollution_multiplier.value = base_pollution_multiplier
```
```suggestion        
```
Looking at this again, I'm not sure what this line is for. What if `pollution_multiplier.value` was previously set to a valid value that is greater than `base_pollution_multiplier`. Do you want to silently change `pollution_multiplier.value` to a different value because someone try to set it to an invalid value?

Or am I missing something?

Jayefuu committed:
Updated pollution multiplier minimum limits 

Following Discord messages with @grilledham, updated the messages to be clearer.

Also updated some broken logic and reduced spam to admins if someone tries to set it below the minimum and it's already at the minimum.

Changes untestested. I'll test tonight or tomorrow if it passes review, not got Factorio on this laptop.

Jayefuu committed:
Fixed failed lua checks

grilledham commented on map_gen/maps/crash_site/outpost_builder.lua:
```diff
@@ -1227,6 +1228,16 @@ local function set_pollution_multiplier(args, player)
+        else -- update the value to the minimum and continue to message all admins
+            player.print("Setting magic crafter pollution multiplier to the minimum value of " .. base_pollution_multiplier)
+            multiplier = base_pollution_multiplier
+        end
```
```suggestion
        end
        -- update the value to the minimum and continue to message all admins
        player.print("Setting magic crafter pollution multiplier to the minimum value of " .. base_pollution_multiplier)
        multiplier = base_pollution_multiplier        
```

grilledham reviewed (approved):
Looks good.

One minor style point, you don't really need the `else` after the `return`.

Jayefuu committed:
Update map_gen/maps/crash_site/outpost_builder.lua

Co-authored-by: grilledham <jrdgillham@gmail.com>

Jayefuu committed:
Fixed LUA check
